src/Estimators/include/BipedalLocomotion/FloatingBaseEstimators/BaseEstimatorFromFootIMU.h

```diff
@@ -0,0 +1,183 @@
+
+            /*
+             * Set the input of the estimator.
+             * @param input the input of the system.
+             * @return true in case of success, false otherwise.
+             */
+            bool setInput(const BaseEstimatorFromFootIMUInput& input) override;
+
+            /*
+             * Perform one step of the estimator.
+             * The estimator must have been initialized first.
+             * @return true in case of success, false otherwise.
+             */
+            bool advance() override;
+
+            /*
+             * Check if the output of the estimator is valid.
+             * @return true in case of success, false otherwise.
+             */
+            bool isOutputValid() const override;
+
+            /*
+             * Get the internal state of the BaseEstimatorFromFootIMU.
+             * @return a const reference to the state of the estimator.
+             */
+            const BaseEstimatorFromFootIMUState& getOutput() const override;
+
+            /*
+             * OPTIONAL
+             * Set the state of the BaseEstimatorFromFootIMU.
+             * @param state of the BaseEstimatorFromFootIMU
+             * @note This method should be called only if the user want to force the state of the system.
+             */
+            void setState(const BaseEstimatorFromFootIMUState& state);
+
+        private:
+            BaseEstimatorFromFootIMUInput m_input; /**< Last input stored in the estimator */
+            BaseEstimatorFromFootIMUState m_state; /**< Current state stored in the estimator */
+            Params m_params;                       /**< Set of possible parameters (not used at the moment) */
+
+            // Geometric quantities of the foot
+            double m_footWidth;
+            double m_footLength;
+
+            // Define the 4 foot vertices in World reference frame
+            Eigen::Vector3d m_p1;
+            Eigen::Vector3d m_p2;
+            Eigen::Vector3d m_p3;
+            Eigen::Vector3d m_p4;
+
+            manif::SO3d m_measuredRotationCorrected;
+            manif::SE3d m_measuredFootPose;
+            manif::SE3d m_frame_H_link;
+
+            Eigen::Vector3d m_gravity;
+
+            iDynTree::KinDynComputations m_kinDyn;
+            iDynTree::Model m_model;
+            iDynTree::LinkIndex m_linkIndex;
+            int m_frameIndex;
+            std::string m_frameName;
+            std::string m_baseFrameName;        //root link of the robot (whose pose must be estimated)
+            StanceFootRF m_stanceFootRF{StanceFootRF::l_sole};
+            bool m_isOutputValid{false};
+            bool m_isModelSet{false};
+            bool m_isStanceFootSet{false};
+            bool m_isInitialized{false};
+        };
+
+    } // namespace Math
+} // namespace BipedalLocomotion
+
+#endif // BIPEDAL_LOCOMOTION_MATH_BASE_ESTIMATION_FROM_FOOT_IMU_H
```

Comment:
Remember to addd a white line at the end of the files
```suggestion
#endif // BIPEDAL_LOCOMOTION_MATH_BASE_ESTIMATION_FROM_FOOT_IMU_H

```